src/components/protocol_handler/src/protocol_packet.cc

```diff
@@ -319,34 +342,56 @@ RESULT_CODE ProtocolPacket::ProtocolHeaderValidator::validate(
   // For Control frames Data Size value shall be less than MTU header
   // For Single and Consecutive Data Size value shall be greater than 0x00
   // and shall be less than payload size
-  if (header.dataSize > payload_size) {
-    LOG4CXX_WARN(logger_,
-                 "Packet data size is "
-                     << header.dataSize
-                     << " and bigger than allowed payload size " << payload_size
-                     << " bytes");
-    return RESULT_FAIL;
-  }
+  // For First Frame Data Size shall be equal 0x08 (payload of this packet will
+  // be 8 bytes).
+
   switch (header.frameType) {
     case FRAME_TYPE_SINGLE:
     case FRAME_TYPE_CONSECUTIVE:
-      if (header.dataSize <= 0u) {
+    case FRAME_TYPE_CONTROL: {
+      bool wrongDataSize = false;
+
+      if (header.dataSize > payload_size) {
+        wrongDataSize = true;
+      }
+
+      if (FRAME_TYPE_CONTROL != header.frameType && header.dataSize <= 0u) {
+        wrongDataSize = true;
+      }
+
+      if (wrongDataSize) {
```

Comment:
Conditions for wrongDataSize aren't complicated enough to warrant more variables. Both conditions can go in this if statement.

Reply:
@iCollin done in [0638eaf](https://github.com/smartdevicelink/sdl_core/pull/2516/commits/0638eafbedee6ca6f9401e6f8aaa2514839507b2)

Reply:
@ydementieiev we really don't need to declare bool at all.

```
if (header.dataSize > payload_size || (FRAME_TYPE_CONTROL != header.frameType && 0u == header.dataSize)) {
```

with check_style.sh of course. Thank you!

Reply:
@iCollin done [8291f1fc](https://github.com/smartdevicelink/sdl_core/pull/2516/commits/8291f1fc155610bd60c00b80d84b10746558a9a4). Thank you